backend/dataall/modules/s3_datasets/cdk/dataset_stack.py

```diff
@@ -437,20 +449,68 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
             },
         )
 
+        glue_sec_conf_enc_key = kms.Key(
+            self,
+            f'crwlr_log_enc_key_{dataset.GlueCrawlerName}',
+            removal_policy=RemovalPolicy.DESTROY,
+            alias=f'crwlr_log_enc_key_{dataset.GlueCrawlerName}',
+            enable_key_rotation=True,
+            admins=[
+                iam.ArnPrincipal(env.CDKRoleArn),
+            ],
+        )
+
+        glue_sec_conf_enc_key.add_to_resource_policy(
+            iam.PolicyStatement(
+                sid='EnableCrawlerIAMRoleKeyUsage',
+                resources=['*'],
+                effect=iam.Effect.ALLOW,
+                principals=[dataset_admin_role],
+                actions=['kms:Encrypt', 'kms:Decrypt', 'kms:ReEncrypt*', 'kms:GenerateDataKey*', 'kms:DescribeKey'],
+            )
+        )
+
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('logs.amazonaws.com'))
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('glue.amazonaws.com'))
+        glue_sec_conf_enc_key.grant_encrypt_decrypt(iam.ServicePrincipal('s3.amazonaws.com'))
+
+        glue_crawler_security_config = glue.CfnSecurityConfiguration(
+            self,
+            f'crwlr_sec_config-{dataset.GlueCrawlerName}',
+            encryption_configuration=glue.CfnSecurityConfiguration.EncryptionConfigurationProperty(
+                cloud_watch_encryption=glue.CfnSecurityConfiguration.CloudWatchEncryptionProperty(
+                    cloud_watch_encryption_mode='SSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                ),
+                job_bookmarks_encryption=glue.CfnSecurityConfiguration.JobBookmarksEncryptionProperty(
+                    job_bookmarks_encryption_mode='CSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                ),
+                s3_encryptions=[
+                    glue.CfnSecurityConfiguration.S3EncryptionProperty(
+                        s3_encryption_mode='SSE-KMS', kms_key_arn=glue_sec_conf_enc_key.key_arn
+                    )
+                ],
+            ),
+            name=f'crwlr_sec_config-{dataset.GlueCrawlerName}',
```

Comment:
- similar to above - can we have both `name` and `glue.CfnSecurityConfiguration` construct id be `{dataset_basename}-security-config` (using basename from above)

   - that way the naming convention is prefixed with `resource_prefix`

- nit: we can also set it once as variable and pass that as both construct id and alias name

Reply:
done